libs/api/ebpf_api.cpp

```diff
@@ -4040,6 +4040,19 @@ ebpf_get_link_fd_by_id(ebpf_id_t id, _Out_ int* fd) NO_EXCEPT_TRY
 }
 CATCH_NO_MEMORY_EBPF_RESULT
 
+_Must_inspect_result_ ebpf_result_t
+ebpf_get_handle_from_fd(int fd, _Out_ intptr_t* handle) NO_EXCEPT_TRY
+{
+    EBPF_LOG_ENTRY();
+    intptr_t fdHandle = intptr_t(-1);
+    if (fdHandle = Platform::_get_osfhandle(fd); fdHandle == intptr_t(-1)) {
```

Comment:
This is a confusing construct, and the fdHandle = intptr_t(-1) part duplicates what is in 4047.  I think you want to compare the result to INVALID_HANDLE_VALUE.

Reply:
Updated